src/syntax_text.rs

```diff
@@ -59,7 +58,7 @@ impl SyntaxText {
 
     pub fn slice<R: private::SyntaxTextRange>(&self, range: R) -> SyntaxText {
         let start = range.start().unwrap_or_default();
-        let end = range.end().unwrap_or(self.len());
+        let end = range.end().unwrap_or_else(|| self.len());
```

Comment:
This also seems like a false-positive to me: everything here should be inalienable and side-effect and allocation free, so I'd be surprised if there are any significant perf differences. OTOH, readability suffers because of a lambda. 

On a more general note, I do appreciate fixing clippy lints (even I don't agree with some particular ones), but it might be prudent to separate "hey, I've run clippy" and "hey, I've completely rewritten the core of this library, It's supper fast now. Oh, and I've added a dozen of `unsafe` blocks" pull requests :D